src/parser/jobs/gnb/modules/Ammo.tsx

```diff
@@ -0,0 +1,149 @@
+import {t, Trans} from '@lingui/macro'
+import Color from 'color'
+import {Fragment} from 'react'
+
+import {ActionLink} from 'components/ui/DbLink'
+import TimeLineChart from 'components/ui/TimeLineChart'
+import ACTIONS from 'data/ACTIONS'
+import JOBS from 'data/JOBS'
+import {CastEvent} from 'fflogs'
+import Module, {dependency, DISPLAY_MODE} from 'parser/core/Module'
+import Checklist, {Requirement, Rule} from 'parser/core/modules/Checklist'
+
+const AMMO_GENERATORS = {
+	[ACTIONS.SOLID_BARREL.id]: 1,
+	[ACTIONS.DEMON_SLAUGHTER.id]: 1,
+	[ACTIONS.BLOODFEST.id]: 2,
+}
+
+const AMMO_SPENDERS = {
+	[ACTIONS.GNASHING_FANG.id]: 1,
+	[ACTIONS.BURST_STRIKE.id]: 1,
+	[ACTIONS.FATED_CIRCLE.id]: 1,
+}
+
+const MAX_AMMO = 2
+
+class AmmoState {
```

Comment:
Given how it's being used, this should probs be an `interface`.
I'm also not seeing any paths where `time` or `ammo` could be `null` at a glance (apologies if I missed something) - so you can probs reduce the type on both of those to just `number`. As an `interface`, you wouldn't need the initialiser, either.

Off-topic for the above, I'll _generally_ prefer `foo?: type` over `foo: type | null`. There's absolutely a bunch of stuff in the existing codebase that uses `null` instead of `undefined`, but I'm generally moving away from that to a consistent `undefined`.